rtt/Service.cpp

```diff
@@ -263,19 +264,8 @@ namespace RTT {
 
         this->mowner = new_owner;
 
-        for( Services::iterator it= services.begin(); it != services.end(); ++it) {
+        for( Services::iterator it= services.begin(); it != services.end(); ++it)
             it->second->setOwner( new_owner );
-            if (new_owner) {
-                // we pass and store a shared ptr in setParent, so we hack it like this:
-                shared_ptr me;
-                try {
-                    me = shared_from_this();
-                } catch ( boost::bad_weak_ptr& bw ) {
-                    me.reset(this); // take ownership
-                }
-                it->second->setParent( me );
-            }
-        }
```

Comment:
Not sure why this section was ever required. When iterating over child services in `Service::setOwner()` their parent pointer should already be set to ourselves (called from `Service::addService()`), so no need to reset it here.